modules/openapi-generator/src/main/resources/dart/libraries/dio/serialization/freezed/api/deserialize.mustache

```diff
@@ -0,0 +1,33 @@
+{{#isResponseFile}}
+    _responseData = _response.data as {{{returnType}}};
+{{/isResponseFile}}
+{{^isResponseFile}}
+    {{#returnSimpleType}}
+        {{#returnTypeIsPrimitive}}
+            _responseData = _response.data as {{{returnType}}};
+        {{/returnTypeIsPrimitive}}
+        {{^returnTypeIsPrimitive}}
+            _responseData = {{{returnType}}}.fromJson(_response.data as Map<String,dynamic>);
+        {{/returnTypeIsPrimitive}}
+    {{/returnSimpleType}}
+    {{^returnSimpleType}}
+        {{#returnTypeIsPrimitive}}
+            _responseData = _response.data as {{{returnType}}};
+        {{/returnTypeIsPrimitive}}
+        {{^returnTypeIsPrimitive}}
+            {{#isArray}}
+                {{#uniqueItems}}
+                    final _responseDataAsSet = _response.data as List<dynamic>;
+                        _responseData = _responseDataAsSet.map<{{{returnBaseType}}}>((dynamic e)=> {{{returnBaseType}}}.fromJson(e as Map<String, dynamic>)).toSet();
+                {{/uniqueItems}}
+                {{^uniqueItems}}
+                        final _responseDataAsList = _response.data as List<dynamic>;
+                        _responseData = _responseDataAsList.map<{{{returnBaseType}}}>((dynamic e)=> {{{returnBaseType}}}.fromJson(e as Map<String, dynamic>)).toList();
```

Comment:
does this handle `List<String/num/etc..>` ?

Reply:
you mean a List<List<String>>?

Reply:
I mean that `e as Map<String, dynamic>` will throw if `e` is a Primitive or an `Iterable`

Reply:
I assumed line 14-16 would protect the case you mentioned. However I will validate this.

Reply:
I assumed line 14-16 would protect the case you mentioned. However I will validate this.